kernel/darwin/iokit/driver.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ *  Copyright (c) 2015, Google, Inc.
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include "kernel/darwin/iokit/driver.h"
+
+#include <IOKit/IODataQueueShared.h>
+#include <kern/clock.h>
+#include <sys/proc.h>
+#include <sys/vnode.h>
+
+#include "kernel/darwin/iokit/public.h"
+#include "kernel/darwin/iokit/logging.h"
+
+// This required macro defines the class's constructors, destructors,
+// and several other methods I/O Kit requires.
+OSDefineMetaClassAndStructors(com_facebook_iokit_OSQuery, IOService)
```

Comment:
nit, we should use `com_facebook_iokit_osquery`. No caps unless osquery is starting a sentence or method name, in which case it'll be `Osquery`.